crypto/ec/ec_curve.c

```diff
@@ -3117,25 +3117,65 @@ static const ec_list_element curve_list[] = {
 
 #define curve_list_length OSSL_NELEM(curve_list)
 
-const char *ec_curve_nid2name(int nid)
+static const ec_list_element *ec_curve_nid2curve(int nid)
 {
     int i;
 
+    if (nid <= 0)
+        return NULL;
+
     for (i = 0; i < (int)curve_list_length; i++) {
```

Comment:
Perhaps just make `i` a `size_t` and get rid of the cast?

Reply:
Done